decotools/fileio_android.py

```diff
@@ -0,0 +1,206 @@
+    data_dir : str, optional
+        Path to directory containing images (default is /net/deco/deco_data).
+
+    Returns
+    -------
+    image_files : str, array-like
+        Image file paths.
+
+    '''
+    date, basename = os.path.split(path)
+    image_id = basename.split('_')[0]
+    image_file = os.path.join(data_dir, date + 'Z', image_id+'.jpg')
+
+    return image_file
+
+
+@np.vectorize
+def db_path_to_date(path, data_dir='/net/deco/deco_data'):
+    '''Function to extract date from paths stored in andriod db
+
+    Parameters
+    ----------
+    path : str, array-like
+        Path from android database.
+    data_dir : str, optional
+        Path to directory containing images (default is /net/deco/deco_data).
+
+    Returns
+    -------
+    dates : str, array-like
+        Dates corresponding to path.
+
+    '''
+    date, basename = os.path.split(path)
+    return pd.to_datetime(date, utc=True)
+
+
+def get_android_files(start_date=None, end_date=None, data_dir='/net/deco/deco_data',
+                      db_file='/net/deco/db_hourly_safe.csv',
+                      include_events=True, include_min_bias=False,
+                      device_id=None, return_metadata=False, verbose=0):
+    '''Function to retrieve deco android image files
+
+    Parameters
+    ----------
+    start_date : str, optional
+        Starting date for the files to retrieve. Use any common
+        date format (e.g. '2017-01-01', '20170101', 'Jan 1, 2017', etc).
+        Default starting date is '2010.01.01'.
+    end_date : str, optional
+        Ending date for the files to retrieve. Use any common
+        date format (e.g. '2017-01-01', '20170101', 'Jan 1, 2017', etc).
+        Default is the current date.
+    data_dir : str, optional
+        Base directory to retrieve android image files.
+    db_file : str, optional
+        File path to android database.
+    include_events : bool, optional
+        Option to include images files flagged as events. Default is True.
+    include_min_bias : bool, optional
+        Option to include minimum bias image files. Default is False.
+    device_id : str or array-like, optional
+        Option to specify which devices you want to look at. Can
+        either be a string, e.g. 'DECO-00000000-450a-7561-433f-0516209b4922',
+        or a list of device IDs, e.g. ['DECO-00000000-450a-7561-433f-0516209b4922',
+        'DECO-ffffffff-bd6f-e5fb-842b-56b10033c587']. Default is to include all
+        device IDs.
+    return_metadata : boolean, optional
+        Return a DataFrame with metadata information for each image file
+        (default is False).
+    verbose : int {0, 1, or 2}
+        Option to have verbose output when getting files. Where 0 is
+        least verbose, while 2 is the most verbose.
+
+    Returns
+    -------
+    numpy.ndarray
+        Numpy array containing files that match specified criteria
+
+    '''
+    # Check that data_dir and db_file exist:
+    if not os.path.isdir(data_dir):
```

Comment:
@jrbourbeau I guess this will automatically take care of any images that have no path in the database?